src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -329,6 +329,34 @@ bool ClusterResourceScheduler::GetNodeResources(int64_t node_id,
 
 int64_t ClusterResourceScheduler::NumNodes() { return nodes_.size(); }
 
+void ClusterResourceScheduler::AddLocalResource(
+                                                      const std::string &resource_name,
+                                                      double resource_total) {
+  string_to_int_map_.Insert(resource_name);
+  int64_t resource_id = string_to_int_map_.Get(resource_name);
+
+  if (local_resources_.custom_resources.find(resource_id) != local_resources_.custom_resources.end()) {
+    FixedPoint total(resource_total);
+    auto &instances = local_resources_.custom_resources[resource_id];
+    instances.total[0] += total;
+    instances.available[0] += total;
+    auto &capacity = nodes_[local_node_id_].custom_resources[resource_id];
+    capacity.available += total;
+    capacity.total += total;
+  } else {
+    ResourceInstanceCapacities capacity;
+    capacity.total.resize(1);
+    capacity.total[0] = resource_total;
+    capacity.available.resize(1);
```

Comment:
This resize looks a little ugly. Any special reason why we don't just use emplace?

Reply:
I think resize handles more edge cases since we want this vector to always have length 1. It's a bit of a weird pattern the first time you see it, but this is also how we initialize resources in other parts of the code (see `InitResourceInstances`).